src/main/java/sentinel/Sentinel.java

```diff
@@ -0,0 +1,247 @@
+
+import java.io.IOException;
+import java.time.LocalDate;
+import java.util.Scanner;
+
+import sentinel.command.Commands;
+import sentinel.storage.Storage;
+import sentinel.task.TaskList;
+import sentinel.ui.Ui;
+
+/**
+ * Sentinel chatbot system.
+ *
+ * @author isaactodo (Isaac Lim Tzee Zac)
+ */
+public class Sentinel {
+    private TaskList taskList;
+    private TaskList searchTaskList;
+    private final Ui ui;
+
+    private boolean isSearch;
+
+    /**
+     * Constructs a new Sentinel and initializes it with commands and
+     * a task list.
+     * The task list is initially empty.
+     */
+    public Sentinel() {
+        this.ui = new Ui();
+        this.isSearch = false;
+
+        // Load save file
+        try {
+            this.taskList = Storage.load();
+        } catch (IOException exception) {
+            ui.showError("OOPS NO FILE???");
+            this.taskList = new TaskList();
+        }
+    }
+
+    ////////////// COMMANDS FOR SENTINEL START //////////////
+    /**
+     * Makes Sentinel output the list of tasks.
+     */
+    public void outputTaskList() {
+        isSearch = false;
+
+        say("Here are the list of your tasks \n" + this.taskList.toString()
+                + String.format("\n You have %d tasks in total.", this.taskList.getTotal()));
+    }
+
+    /**
+     * Makes Sentinel output the list of tasks that match the search string.
+     */
+    public void outputMatchingTaskList(String searchTerm) {
+        searchTaskList = this.taskList.getMatching(searchTerm);
+        isSearch = true;
+
+        say("Here are the list of your matching tasks \n" + this.searchTaskList.toString()
+                + String.format("\n You have %d tasks matching that name.", this.searchTaskList.getTotal())
+                + "\n\n You are now indexing the search list.\n Type list to back to indexing the original list.");
+    }
+
+    /**
+     * Makes Sentinel mark the given task number as done.
+     *
+     * @param taskNumber sentinel.task.Task number to be marked.
+     */
+    public void markDone(int taskNumber) throws SentinelException {
+        TaskList selectedTaskList = this.taskList;
+
+        if (this.isSearch) {
+            selectedTaskList = this.searchTaskList;
+        }
+
+        say("Marked the following task as done: \n " + selectedTaskList.markAsDone(taskNumber));
+    }
+
+    /**
+     * Makes Sentinel mark the given task number as undone.
+     *
+     * @param taskNumber sentinel.task.Task number to be unmarked.
+     */
+    public void markUndone(int taskNumber) throws SentinelException {
+        TaskList selectedTaskList = this.taskList;
+
+        if (this.isSearch) {
+            selectedTaskList = this.searchTaskList;
+        }
+
+        say("Unmarked the following task: \n " + selectedTaskList.markAsUndone(taskNumber));
+    }
+
+    /**
+     * Makes Sentinel add the given todo to the task list.
+     *
+     * @param description Description of the todo to be added.
```

Comment:
Might be some spelling error on "description Description" perchance, same applies to line 107 & 116